scene/gui/code_edit.cpp

```diff
@@ -1291,15 +1291,22 @@ bool CodeEdit::is_line_numbers_zero_padded() const {
 
 void CodeEdit::_line_number_draw_callback(int p_line, int p_gutter, const Rect2 &p_region) {
 	String fc = TS->format_number(String::num(p_line + 1).lpad(line_number_digits, line_number_padding));
-	Ref<TextLine> tl;
-	tl.instantiate();
-	tl->add_string(fc, font, font_size);
-	int yofs = p_region.position.y + (get_line_height() - tl->get_size().y) / 2;
+	TextLine tl;
+	tl.add_string(fc, font, font_size);
+
+	if (p_line == 0) {
+		// cache line height
+		line_height = get_line_height();
+		// this is surprisingly expensive to check so cache it
+		line_text_y_cache = tl.get_size_fast().y;
```

Comment:
Will this not cause weird vertical alignment if we use the height of `0` for all other numbers?

Reply:
the value never differs it's always the same for each line regardless of contents

Reply:
To prove this:
```
	int line_height = get_line_height();
	print_verbose("line height: " + itos(line_height));
	int yofs = p_region.position.y + (line_height - tl->get_size().y) / 2;
```

Reply:
removed from patch